server/src/main/java/org/elasticsearch/index/codec/vectors/es92/ES92BFloat16FlatVectorsWriter.java

```diff
@@ -0,0 +1,482 @@
+            state.segmentInfo.name,
+            state.segmentSuffix,
+            ES92BFloat16FlatVectorsFormat.META_EXTENSION
+        );
+
+        String vectorDataFileName = IndexFileNames.segmentFileName(
+            state.segmentInfo.name,
+            state.segmentSuffix,
+            ES92BFloat16FlatVectorsFormat.VECTOR_DATA_EXTENSION
+        );
+
+        boolean success = false;
+        try {
+            meta = state.directory.createOutput(metaFileName, state.context);
+            vectorData = state.directory.createOutput(vectorDataFileName, state.context);
+
+            CodecUtil.writeIndexHeader(
+                meta,
+                ES92BFloat16FlatVectorsFormat.META_CODEC_NAME,
+                ES92BFloat16FlatVectorsFormat.VERSION_CURRENT,
+                state.segmentInfo.getId(),
+                state.segmentSuffix
+            );
+            CodecUtil.writeIndexHeader(
+                vectorData,
+                ES92BFloat16FlatVectorsFormat.VECTOR_DATA_CODEC_NAME,
+                ES92BFloat16FlatVectorsFormat.VERSION_CURRENT,
+                state.segmentInfo.getId(),
+                state.segmentSuffix
+            );
+            success = true;
+        } finally {
+            if (success == false) {
+                IOUtils.closeWhileHandlingException(this);
+            }
+        }
+    }
+
+    @Override
+    public FlatFieldVectorsWriter<?> addField(FieldInfo fieldInfo) throws IOException {
+        FieldWriter<?> newField = FieldWriter.create(fieldInfo);
+        fields.add(newField);
+        return newField;
+    }
+
+    @Override
+    public void flush(int maxDoc, Sorter.DocMap sortMap) throws IOException {
+        for (FieldWriter<?> field : fields) {
+            if (sortMap == null) {
+                writeField(field, maxDoc);
+            } else {
+                writeSortingField(field, maxDoc, sortMap);
+            }
+            field.finish();
+        }
+    }
+
+    @Override
+    public void finish() throws IOException {
+        if (finished) {
+            throw new IllegalStateException("already finished");
+        }
+        finished = true;
+        if (meta != null) {
+            // write end of fields marker
+            meta.writeInt(-1);
+            CodecUtil.writeFooter(meta);
+        }
+        if (vectorData != null) {
+            CodecUtil.writeFooter(vectorData);
+        }
+    }
+
+    @Override
+    public long ramBytesUsed() {
+        long total = SHALLOW_RAM_BYTES_USED;
+        for (FieldWriter<?> field : fields) {
+            total += field.ramBytesUsed();
+        }
+        return total;
+    }
+
+    private void writeField(FieldWriter<?> fieldData, int maxDoc) throws IOException {
+        // write vector values
+        long vectorDataOffset = vectorData.alignFilePointer(Float.BYTES);
+        switch (fieldData.fieldInfo.getVectorEncoding()) {
+            case BYTE -> writeByteVectors(fieldData);
```

Comment:
Since now we are creating this with `element_type`, we should just throw if `byte` is used when creating a field and remove all the byte support here on the writer & on the reader.